bundle/edu.gemini.spModel.pio/src/main/scala/edu/gemini/spModel/pio/codec/ParamCodec.scala

```diff
@@ -0,0 +1,59 @@
+package edu.gemini.spModel.pio.codec
+
+import scalaz._, Scalaz._
+
+import edu.gemini.spModel.pio._
+import edu.gemini.spModel.pio.xml.{ PioXmlUtil, PioXmlFactory }
+
+trait ParamCodec[A] { outer =>
+  def encode(key: String, a: A): Param
+  def decode(p: Param): PioError \/ A
+  def xmap[B](f: A => B, g: B => A): ParamCodec[B] =
+    new ParamCodec[B] {
+      def encode(key: String, a: B): Param = outer.encode(key, g(a))
+      def decode(p: Param): PioError \/ B = outer.decode(p).map(f)
+    }
+}
+
+object ParamCodec {
+  
+  private val pf = new PioXmlFactory
+
+  def apply[A](implicit ev: ParamCodec[A]) = ev
+
+  implicit val StringParamCodec: ParamCodec[String] =
+    new ParamCodec[String] {
+      def encode(key: String, a: String): Param = pf.createParam(key) <| (_.setValue(a))
+      def decode(p: Param): PioError \/ String = Option(p.getValue) \/> NullValue(p.getName)
```

Comment:
It might not be the case for targets, but sometimes parameters are optional and the absence of a particular parameter isn't an error.  I suppose instead of `String` params in that case they should be `Option[String]` params.